plugins/chat/app/serializers/chat/base_channel_membership_serializer.rb

```diff
@@ -7,6 +7,19 @@ class BaseChannelMembershipSerializer < ApplicationSerializer
                :notification_level,
                :chat_channel_id,
                :last_read_message_id,
-               :last_viewed_at
+               :last_viewed_at,
+               :starred
+
+    def starred
+      if scope.user&.upcoming_change_enabled?(:star_chat_channels)
+        object.starred
+      else
+        false
+      end
+    end
+
+    def include_starred?
+      true
```

Comment:
```suggestion
      authenticated?
```

Since we don't need this for anon users, then your `starred` doesn't have to safe navigate `scope.user`